client/src/app/all-projects/all-projects.component.spec.ts

```diff
@@ -25,10 +25,23 @@ describe('AllProjectsComponent', () => {
     expect(component).toBeTruthy();
   });
 
+  it('empty searchText should hide clear button', () => {
+    expect(component.searchText).toBe('');
+    expect(fixture.nativeElement.querySelector('button')).toBeFalsy();
+  });
+
+  it('value in searchText should make clear button visible', () => {
+    component.searchText = 'sample search text';
+    fixture.detectChanges();
+    expect(fixture.nativeElement.querySelector('button')).toBeTruthy();
+  });
+
   it('should call onClearSearch', fakeAsync(() => {
     spyOn(component, 'onClearSearch');
-    const button = fixture.nativeElement.querySelector('button');
-    button.click();
+    component.searchText = 'sample search text';
+    fixture.detectChanges();
+    const clearButton = fixture.nativeElement.querySelector('button');
+    clearButton.click();
```

Comment:
This should check that searchText is empty and that the value of the input field is empty too

Reply:
Oh my bad! I didn't see the test below.